### DISCUSSION TITLE
GABAb tau doesn't update after changing values

### DISCUSSION THREAD

**Discussion by wagdy88** [jonescompneurolab/hnn-core#625] [Q&A] [answered]
Hi guys,
There seems to be an issue with the GABAb tau function. After changing the values, it still yields identical values. 

Here is the code:
``python
import matplotlib.pyplot as plt
import numpy as np
import hnn_core
from hnn_core import  jones_2009_model, simulate_dipole
net = jones_2009_model(add_drives_from_params= True)
simulate_dipole(net, tstop = 100, record_isec= 'all')
isec = net.cell_response.isec[0][36]
myarray= np.array(isec['soma']['soma_gabab'])]
       

#changed the tau of gabab
myarray1= np.array(isec['soma']['soma_gabab'])

myfig, mysp = plt.subplots(1, 1)
isec1 = np.loadtxt('/home/mohamed/Dropbox/projects/hnn_MMN/results/isec1.npy')
isec2=  np.loadtxt('/home/mohamed/Dropbox/projects/hnn_MMN/results/isec2.npy')
mysp.plot(isec1, color= 'r', label='LongGABAb')
mysp.plot(isec2, color= 'b', label='ZeroGABAb')
mysp.legend()
``
![2023March12_gababb_a](https://user-images.githubusercontent.com/37229661/224562743-39a0391a-a509-45fe-b4d5-4f6b719a1d34.png)

**Comment by ntolley** [comment, marked as answer]
@wagdy88 I'm a bit confused do you mind clarifying? ` myarray1= np.array(isec['soma']['soma_gabab'])` stores the somatic current recorded during the simulation under the variable `myarray1`.

If you want to update the properties of the GABAb receptors you can do it like this:
```py
from hnn_core import  jones_2009_model, simulate_dipole
from hnn_core.viz import plot_dipole

net_original = jones_2009_model(add_drives_from_params=True)
net_new = net_original.copy()

# Here is how you update receptor properties
net_new.cell_types['L5_pyramidal'].synapses['gabab']['tau2'] = 60.0

dpl_original = simulate_dipole(net_original, tstop=170, dt=0.5)
dpl_new = simulate_dipole(net_new, tstop=170, dt=0.5)

plot_dipole([dpl_original[0], dpl_new[0]])
```
![image](https://user-images.githubusercontent.com/55253912/224569929-ab02ca47-0743-4b36-960b-a2e602d890b1.png)

**Comment by jasmainak** [comment]
Btw @wagdy88 you missed one back-tick that's why your code didn't render properly ;-)

**Reply by wagdy88** [reply]
That's why I needed the experts opinion :-)
